specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json

```diff
@@ -0,0 +1,2571 @@
+      "description": "Description of the response of the verification code.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of certificate.",
+          "readOnly": true
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Request etag."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "type": "object",
+          "properties": {
+            "verificationCode": {
+              "type": "string",
+              "description": "Verification code."
+            },
+            "subject": {
+              "type": "string",
+              "description": "Certificate subject."
+            },
+            "expiry": {
+              "type": "string",
+              "description": "Code expiry."
+            },
+            "thumbprint": {
+              "type": "string",
+              "description": "Certificate thumbprint."
+            },
+            "isVerified": {
+              "type": "boolean",
+              "description": "Indicate if the certificate is verified by owner of private key."
+            },
+            "certificate": {
+              "description": "base-64 representation of X509 certificate .cer file or just .pem file content.",
+              "type": "string",
+              "format": "byte"
+            },
+            "created": {
+              "type": "string",
+              "description": "Certificate created time."
+            },
+            "updated": {
+              "type": "string",
+              "description": "Certificate updated time."
+            }
+          }
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "VerificationCodeRequest": {
+      "type": "object",
+      "description": "The JSON-serialized leaf certificate",
+      "properties": {
+        "certificate": {
+          "description": "base-64 representation of X509 certificate .cer file or just .pem file content.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "subscriptionId": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "The subscription identifier.",
+      "required": true,
+      "type": "string"
+    },
+    "api-version": {
+      "name": "api-version",
+      "in": "query",
+      "description": "The version of the API.",
+      "required": true,
+      "type": "string"
+    },
+    "resourceGroupName": {
+      "name": "resourceGroupName",
+      "description": "The name of the resource group that contains the provisioning service.",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
```

Comment:
Refactoring.

Common parameters best practice - use common types
There are definitions of common parameters in common-types/v3/types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter,
See
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630)

This should be non-breaking if you are following the normal pattern, but may look breaking if your definitions of the parameters were somehow nonstandard. In your case if you're going from preview -> GA then I think you should preferably take that breaking change now.